src/quote/getQuoteAndBuyCoverInputs.ts

```diff
@@ -0,0 +1,161 @@
+  GetQuoteResponse,
+  IntString,
+  Integer,
+  PoolCapacity,
+} from '../types';
+
+dotenv.config();
+
+type CoverRouterQuoteParams = {
+  productId: Integer;
+  amount: IntString;
+  period: Integer;
+  coverAsset: CoverAsset;
+};
+
+async function getQuoteAndBuyCoverInputs(
+  productId: Integer,
+  coverAmount: IntString,
+  coverPeriod: Integer, // days
+  coverAsset: CoverAsset,
+  coverBuyerAddress: Address,
+  paymentAsset: CoverAsset = coverAsset,
+  slippage: number = DEFAULT_SLIPPAGE, // 0.1%
+  ipfsData: string = '', // IPFS base32 hash value
+): Promise<GetQuoteApiResponse | ErrorApiResponse> {
+  let result: GetQuoteResponse | undefined = undefined;
+
+  if (!process.env.COVER_ROUTER_URL) {
+    return { result, error: { message: 'Missing COVER_ROUTER_URL env var', data: {} } };
+  }
+
+  if (!Number.isInteger(productId) || productId <= 0) {
+    return { result, error: { message: 'Invalid productId: must be a positive integer', data: {} } };
+  }
+
+  if (!/^\d+$/.test(coverAmount) || parseInt(coverAmount, 10) <= 0) {
+    return { result, error: { message: 'Invalid coverAmount: must be a positive integer string', data: {} } };
+  }
+
+  if (!Number.isInteger(coverPeriod) || coverPeriod < MINIMUM_COVER_PERIOD) {
+    return {
+      result,
+      error: { message: `Invalid coverPeriod: must be at least ${MINIMUM_COVER_PERIOD} days`, data: {} },
+    };
+  }
+
+  if (!Object.values(CoverAsset).includes(coverAsset)) {
+    return { result, error: { message: 'Invalid coverAsset', data: {} } };
+  }
+
+  if (!Object.values(CoverAsset).includes(paymentAsset)) {
+    return { result, error: { message: 'Invalid paymentAsset', data: {} } };
+  }
+
+  if (!/^0x[a-fA-F0-9]{40}$/.test(coverBuyerAddress)) {
+    return { result, error: { message: 'Invalid coverBuyerAddress: must be a valid Ethereum address', data: {} } };
+  }
+
+  if (typeof slippage !== 'number' || slippage < 0 || slippage > 1) {
+    return { result, error: { message: 'Invalid slippage: must be a number between 0 and 1', data: {} } };
+  }
+
+  if (ipfsData !== '' && !/^[A-Za-z2-7]{46}$/.test(ipfsData)) {
+    return { result, error: { message: 'Invalid ipfsData: must be a valid IPFS base32 hash value', data: {} } };
+  }
+
+  try {
+    const { quote, capacities } = await getQuote(productId, coverAmount, coverPeriod, coverAsset);
+
+    const maxPremiumInAsset = calculatePremiumWithCommissionAndSlippage(
+      BigInt(quote.premiumInAsset),
+      DEFAULT_COMMISSION_RATIO,
+      slippage,
+    );
+
+    result = {
+      displayInfo: {
+        premiumInAsset: quote.premiumInAsset,
+        coverAmount,
+        yearlyCostPerc: (BigInt(quote.annualPrice) / BigInt(TARGET_PRICE_DENOMINATOR)).toString(),
+        maxCapacity: getMaxCapacity(capacities),
+      },
+      buyCoverInput: {
+        buyCoverParams: {
+          coverId: CoverId.BUY,
+          owner: coverBuyerAddress,
+          productId,
+          coverAsset,
+          amount: coverAmount,
```

Comment:
Amount value should be in wei e.g. for 1 ETH we should have amount value as `1000000000000000000`. It should be converted first before passing it (both to quote endpoint, and to buyCover contract fn). 

On FE we use `parseEther` from `viem` package to do that - you can check out the code [here](https://github.com/NexusMutual/frontend-next/blob/eed2786805cfb438bb66f46e5143050c8939941b/src/services/covers.service.ts#L166)

Reply:
yes good catch. I was used to taking in all parameters is wei format already in the BE. but here, I guess it will be in the "natural" currency (i.e. ETH)

Reply:
if it comes from user input, it's probably gonna be "natural" currency unless the devs do the conversions before using the sdk

Reply:
as discussed in with @mixplore in slack, since the SDK is aimed for use with devs all inputs and outputs will be in the smallest currency (i.e. wei)